zebra-network/src/peer_set/set.rs

```diff
@@ -41,6 +41,53 @@
 //! [finagle]: https://twitter.github.io/finagle/guide/Clients.html#power-of-two-choices-p2c-least-loaded
 //! [p2c]: http://www.eecs.harvard.edu/~michaelm/postscripts/handbook2001.pdf
 //! [tower-balance]: https://crates.io/crates/tower-balance
+//!
+//! # Behavior During Network Upgrades
+//!
+//! [ZIP-201] specifies peer behavior during network upgrades:
+//!
+//! > With scheduled network upgrades, at the activation height, nodes on each consensus branch
+//! > should disconnect from nodes on other consensus branches and only accept new incoming
+//! > connections from nodes on the same consensus branch.
+//!
+//! Zebra handles this with the help of [`MinimumPeerVersion`], which determines the minimum peer
+//! protocol version to accept based on the current best chain tip height. The minimum version is
+//! therefore automatically increased when the block height reaches a network upgrade's activation
+//! height. The helper type is then used to:
+//!
+//! - cancel handshakes to outdated peers, in
+//!   [`handshake::negotiate_version`][crate::peer::handshake::negotiate_version]
+//! - cancel requests to and disconnect from peers that have become outdated, in
+//!   [`PeerSet::push_unready`]
+//! - disconnect from peers that have just responded and became outdated, in
+//!   [`PeerSet::poll_unready`]
+//! - disconnect from idle peers that have become outdated, in
+//!   [`PeerSet::disconnect_from_outdated_peers`]
+//!
+//! ## Network Coalescence
+//!
+//! [ZIP-201] also specifies how Zcashd behaves [leading up to a activation
+//! height][network-coalescence]. Since Zcashd limits the number of connections to at most eight
+//! peers, it will gradually migrate its connections to up-to-date peers as it approaches the
+//! activation height.
+//!
+//! The motivation for this behavior is to avoid an abrupt partitioning the network, which can lead
+//! to isolated peers and increases the chance of an eclipse attack on some peers of the network.
+//!
+//! Zebra does not gradually migrate its peers as it approaches an activation height. This is
+//! because Zebra by default has a target of connecting to 50 peers, as can be seen in
+//! [`Config::default`]. Since this is a lot larger than the 8 peers Zcashd connects to, an eclipse
```

Comment:
Nit: the default `Config.peerset_initial_target_size` is 50, the default `Config::peerset_outbound_connection_limit` is 75, and the actual number of connected peers varies between 0-75 based on peer set load.

But there are multiple tasks that contact peers every minute or so (peer crawler, mempool crawler, syncer, block & transaction gossipers). So Zebra tends to keep 75 connections open. (And since we've stopped disconnecting from peers due to message order differences, those connections stay open for a long time.)

This might be worth another paragraph - but here's a minimal suggestion:

```suggestion
//! because Zebra by default has a target of connecting to 50-75 peers, as can be seen in
//! [`Config::default`]. Since this is a lot larger than the 8 peers Zcashd connects to, an eclipse
```

Reply:
Done, I tweaked the paragraph a bit. Let me know what you think.